lib/PGObject.pm

```diff
@@ -204,26 +207,34 @@ sub function_info {
         push @queryargs, $args{argschema};
     }
 
-    my $sth = $dbh->prepare($query) || die $!;
-    $sth->execute(@queryargs) || die $dbh->errstr . ": " . $query;
+    my $sth = $dbh->prepare($query) || die PGObject::Util::DBException->new(
+                                            $dbh, $query, @queryargs);
+    $sth->execute(@queryargs) || die PGObject::Util::DBException->new(
+                                            $dbh, $query, @queryargs);
     my $rows = $sth->rows;
     if ($rows > 1) {
         if ($args{argtype1}) {
-            croak $log->fatalf(
-                'Ambiguous criteria discovering function %s.%s (with first argument type %s)',
-                $args{funcschema}, $args{funcname}, $args{argtype1}
-                );
+            my $e = PGObject::Util::DBException->internal(_dup_func,
+                    sprintf('Ambiguous criteria discovering function %s.%s (with first argument type %s)',
+                        $args{funcschema}, $args{funcname}, $args{argtype1}
+                    ), $query, @queryargs);
```

Comment:
Does it make sense to include the returned rows in the response? As far as I'm aware the results already exist in memory (because DBD::Pg loads the entire resultset during `execute()`). So including the resultset doesn't add memory pressure, I'd think.

Reply:
I am not quite sure how that would work though.  Should we do a comma-separated list including all arguments?   Or one per line?

Reply:
I was thinking that the exception object can contain the row, maybe without serializing it, but at least so that the caller has access to it?